src/components/App/App.js

```diff
@@ -24,37 +25,74 @@ class App extends Component {
   }
 
   addPokemon = (queriedPokemon) => {
-    this.validatePokemonData(queriedPokemon)
+    const foundPokemon = this.validatePokemonData(queriedPokemon)
+    this.setState({
+      foundPokemon: [foundPokemon]
+    })
     //set state with the validated pokemon
   }
 
   validatePokemonData = (queriedPokemon) => {
     const lowerCaseInput = queriedPokemon.queriedPokemon.toLowerCase()
-
-    const verifiedName = this.state.pokeDex.find(pokemon => {
+    
+    const verifiedName = this.state.pokeDex.find((pokemon, index) => {
       let lowerCaseName = pokemon.name.toLowerCase()
-      if (lowerCaseName.includes(lowerCaseInput) && lowerCaseInput !== '')  {
+
+      if (lowerCaseName.includes(lowerCaseInput) && lowerCaseInput !== '' && lowerCaseInput !== undefined) {
         return pokemon
+      } else if (parseInt(lowerCaseInput)){
+        if (index+1 === parseInt(lowerCaseInput)){
+          return pokemon
+        }
       }
-    })
+    });
+
+    console.log(verifiedName)
 
     if (verifiedName === undefined) {
-      console.log('No Good NAMe!')
-    } else {
+      return this.setState({ error: "Not  a valid name, try again" })
```

Comment:
We can change this to be either not a valid 'name' or 'id'

Reply:
Yes, I'm excited to merge in the NaN function to differentiate this error.